js/listdates-min.js

```diff
@@ -58,21 +118,25 @@ xhr.onload = function() {
 
     // We got a good response from the server
     if (xhr.status >= 200 && xhr.status < 400) {
-        const dates = JSON.parse(xhr.responseText);
-        document.getElementById('datelist-content').innerHTML = listDates(dates);
+
+        // parse the dates
+        dates = parseDates(JSON.parse(xhr.responseText));
+
+        // print the dates
+        datelistContent.innerHTML = buildHTML(dates);
 
     // We reached our target server, but it returned an error
     } else {
         const error = 'Whoops! Something went wrong. Please try again.';
-        document.getElementById('datelist-error').innerHTML = error;
+        datelistError.innerHTML = error;
     }
 };
 
 // Handle an Ajax connection error
 xhr.onerror = function() {
     'use strict';
     const error = 'Whoops! We couldn\'t reach the server. Please try again.';
-    document.getElementById('datelist-error').innerHTML = error;
+    datelistError.innerHTML = error;
```

Comment:
I like that you hoisted these to the top; it was really easy to understand since I had already seen them being defined at the front.